packages/ui/src/forum/types/ForumCategory.ts

```diff
@@ -13,15 +17,37 @@ export interface ForumModerator {
   avatar?: string
 }
 
-interface ForumSubCategory {
+export interface ForumSubCategory {
   id: string
   title: string
 }
 
+export type ForumBreadcrumb = ForumSubCategory
+
 export const asForumCategory = (fields: Omit<ForumCategoryFieldsFragment, '__typename'>): ForumCategory => ({
   id: fields.id,
   title: fields.title,
   description: fields.description,
   subcategories: [],
   moderators: [],
 })
+
+const asSubCategory = (fields: ForumSubCategoryFieldsFragment): ForumBreadcrumb => ({
+  id: fields.id,
+  title: fields.title,
+})
+
+export const asForumBreadcrumbs = (fields: ForumCategoryBreadcrumbsFieldsFragment): ForumBreadcrumb[] => {
+  const breadcrumbs: ForumBreadcrumb[] = []
+  assignBreadcrumbs(fields, breadcrumbs)
+  return breadcrumbs
+}
```

Comment:
This looks like something that could be avoided as `assingBreadcrumbs()` is not "in the spirit of functional programming" (IMO). It does side-effect to the passed params.

This could be written using a loop or `Array.reduce()` since it is a linear structure.

```typescript
return fields.reduce((acc, item) => {
  if (item.parent) {
    acc.push(asSubCategory(item.parent)) //.unshift()?
  }
  return acc
}, [] )
```

Reply:
I decided to still use a recursive function (that no longer pushes elements to an external array) as it works very well for inserting elements in a reverse order and the depth is predetermined by query depth